packages/esm-patient-registration-app/src/patient-registration/input/custom-input/identifier/identifier-input.component.tsx

```diff
@@ -43,6 +43,29 @@ const IdentifierInput: React.FC<IdentifierInputProps> = ({ patientIdentifier, fi
     return map;
   }, [defaultPatientIdentifierTypes]);
 
+  const validateInput = (value: string) => {
+    if (!value || value === '') {
+      return;
+    }
+
+    if (!identifierType?.format) {
+      return;
+    }
+
+    try {
+      const regex = new RegExp(identifierType.format);
+      if (regex.test(value)) {
+        return;
+      }
+      return t('invalidIdentifierFormat', '{{description}}', {
```

Comment:
The only thing I question about this is this part of the PR, specifically using the translation function here. I'm struggling to see who can leverage this. On the one hand, no one can do anything useful with translating this in Transifex since it's just string interpolation. Presumably, an implementation could override the translation, but even then, they'd want something that varies depending on the identifier type.

I think for this one, we just return the `identifierType.formatDescription` and defer translating that to the backend. We could then probably provide a useful frontend translation for the fallback text, namely, `Expected format: {{identifierTypeFormat}}`.